src/perform_step/extrapolation_perform_step.jl

```diff
@@ -142,19 +143,34 @@ function perform_step!(integrator,cache::AitkenNevilleConstantCache,repeat_step=
       end
     end
   else
-    Threads.@threads for i in 2:min(size(T)[1], cur_order+1)
-      dt_temp = dt/(2^(i-1))
-      # Solve using Euler method
-      @muladd u_temp = @.. uprev + dt_temp*integrator.fsalfirst
-      k_temp = f(u_temp, p, t+dt_temp)
 
-      for j in 2:2^(i-1)
-        @muladd u_temp = @.. u_temp + dt_temp*k_temp
-        k_temp = f(u_temp, p, t+j*dt_temp)
+    # Balance workload of threads by computing T[1,1] with T[max_order,1] on
+    # same thread, T[2,1] with T[max_order-1,1] on same thread. Similarly fill
+    # first column of T matrix
+    Threads.@threads for i in 1:ceil(max_order/2)
+      # indices array hold row index of T matrix to be computed in this
+      # execution of thread
+      indices = [Int(i)]
```

Comment:
this will allocate more than you want. Re-structure to not have this array alloc.

Reply:
Should we do the calculation of `T[i,1]` and `T[max_order+1-i,1]` by copy-pasting the same code twice? That will reduce the memory allocation though. Is there any other way around?

Reply:
You can make a tuple, but remember that its length is then type information, so always make it length 2 and check for equality.